haystack/components/generators/azure.py

```diff
@@ -0,0 +1,146 @@
+import logging
+from typing import Optional, Callable, Dict, Any
+
+# pylint: disable=import-error
+from openai.lib.azure import AzureADTokenProvider, AzureOpenAI
+
+from haystack import default_to_dict, default_from_dict
+from haystack.components.generators import OpenAIGenerator
+from haystack.components.generators.utils import serialize_callback_handler, deserialize_callback_handler
+from haystack.dataclasses import StreamingChunk
+
+logger = logging.getLogger(__name__)
+
+
+class AzureGenerator(OpenAIGenerator):
```

Comment:
As the client is referred to as `AzureOpenAI` I wonder whether `AzureOpenAIGenerator` is more intuitive than just `AzureGenerator`. `AzureGenerator` is shorter but I'd prefer the more explicit `AzureOpenAIGenerator` and even the longer `AzureOpenAIChatGenerator`.